docs/synapse/userguides/storm_adv_functions.rstorm

```diff
@@ -0,0 +1,708 @@
+way the pipeline affects your **working set.**
+
+A function in Storm has its **own node pipeline,** independent of any Storm logic that invokes the function.
+Functions **do not** inherit the pipeline of the invoking query, and **do not** modify the invoking pipeline
+by default. 
+
+Nodes are not "sent inbound" to a function to "cause" it to execute; a function runs when it is invoked.
+It is possible to **invoke** a function so that it operates on a node or nodes; but the function will not
+"automatically" do so.
+
+Because the function **itself** is a Storm pipeline, all caveats about pipelines and awareness of your
+working set still apply to the Storm **within** the function.
+
+All of Storm's features and capabilities are available for use within a function. This includes Storm
+operations, commands, `variables`_, `methods`_, `control flow`_, `libraries`_, etc. `Init blocks`_ and
+`fini blocks`_ can be used within functions to execute a subset of Storm logic a single time before
+operating on any nodes in the functions's pipeline or after any nodes have exited the function's pipeline,
+respectively.
+
+.. _storm-func-basics:
+
+Function Basics
+---------------
+
+Storm supports three types of functions. Each is explained in more detail below.
+
+- **callable** functions, which are "regular" functions (similar to functions in other programming languages).
+  Callable functions return a value.
+- **data emitter** functions, which emit data.
+- **node yielder** functions, which yield nodes.
+
+.. _storm-func-declare:
+
+Declaring Functions
++++++++++++++++++++
+
+All functions are **declared** in the same way, using the ``function`` keyword.
+
+- ``function myFunction() { <do stuff> }``
+- ``function myFunction(foo) { <do stuff> }``
+- ``function myFunction(bar,baz=$lib.null) { <do stuff> }``
+
+.. _storm-func-invoke:
+
+Invoking Functions
+++++++++++++++++++
+
+All functions are **invoked** using the function name preceded by the dollar sign ( ``$`` ), and by
+passing any required parameters to the function.
+
+- ``$myFunction()``
+- ``$myFunction(foo)``
+- ``$myFunction($foo)``
+- ``$myFunction(bar,baz=wheee)``
+
+Parameters can be passed as literals, variables, or keyword=$valu arguments. For example, given a
+function that takes an organization name as input:
+
+::
+  
+  function $myFunction(orgname) { <do stuff> }
+
+The name can be passed directly:
+
+::
+  
+  $myFunction("The Vertex Project")
+
+...as a variable:
+
+::
+  
+  $name="The Vertex Project"
+  $myFunction($name)
+
+...or as a keyword/value pair:
+
+::
+  
+  $myFunction(orgname="The Vertex Project")
+
+::
+  
+  $name="The Vertex Project"
+  $myFunction(orgname=$name)
+
+
+Operating on Nodes
+******************
+
+Functions do not inherit or operate on the invoking Storm pipeline by default. If you want a function
+to operate on nodes in a pipeline, you can invoke the function by passing a property (or properties)
+of the node as input to the function.
```

Comment:
You can also pass in the full `$node` and then optionally yield that parameter to get the node into your function's pipeline.

Reply:
revised